clients/src/test/java/org/apache/kafka/clients/producer/internals/RecordAccumulatorTest.java

```diff
@@ -1737,30 +1735,48 @@ private RecordAccumulator createTestRecordAccumulator(int deliveryTimeoutMs, int
      * Return a test RecordAccumulator instance
      */
     private RecordAccumulator createTestRecordAccumulator(
-        TransactionManager txnManager,
-        int deliveryTimeoutMs,
-        int batchSize,
-        long totalSize,
-        Compression compression,
-        int lingerMs
+            TransactionManager txnManager,
+            int deliveryTimeoutMs,
+            int batchSize,
+            long totalSize,
+            Compression compression,
+            int lingerMs
     ) {
         long retryBackoffMs = 100L;
         long retryBackoffMaxMs = 1000L;
         String metricGrpName = "producer-metrics";
 
         return new RecordAccumulator(
-            logContext,
-            batchSize,
-            compression,
-            lingerMs,
-            retryBackoffMs,
-            retryBackoffMaxMs,
-            deliveryTimeoutMs,
-            metrics,
-            metricGrpName,
-            time,
-            new ApiVersions(),
-            txnManager,
-            new BufferPool(totalSize, batchSize, metrics, time, metricGrpName));
+                logContext,
+                batchSize,
+                compression,
+                lingerMs,
+                retryBackoffMs,
+                retryBackoffMaxMs,
+                deliveryTimeoutMs,
+                metrics,
+                metricGrpName,
+                time,
+                new ApiVersions(),
+                txnManager,
+                new BufferPool(totalSize, batchSize, metrics, time, metricGrpName)) {
+            @Override
+            protected BuiltInPartitioner createBuiltInPartitioner(LogContext logContext, String topic,
+                                                                  int stickyBatchSize) {
+                return new MockRandomBuiltInPartitioner(logContext, topic, stickyBatchSize);
+            }
+        };
+    }
+
+    private class MockRandomBuiltInPartitioner extends BuiltInPartitioner {
```

Comment:
it can be static class, and it would be great to unify the naming -> `SequentialPartitioner`

Reply:
It needs to access the instance properties of the test class so can't be static class